src/Fixer/Comment/SingleLineCommentStyleFixer.php

```diff
@@ -78,28 +112,30 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
             if (!$token->isGivenKind(T_COMMENT)) {
                 continue;
             }
+
             $content = $token->getContent();
             $commentContent = substr($content, 2, -2);
             if ($this->hashEnabled && '#' === $content[0]) {
                 $tokens[$index] = new Token([$token->getId(), '//'.substr($content, 1)]);
                 continue;
             }
-            if (!$this->asteriskEnabled || '/*' !== substr($content, 0, 2) || preg_match('/[^\s\*].*\R.*[^\s\*]/s', $commentContent)) {
+            if (!$this->asteriskEnabled || '/*' !== substr($content, 0, 2) || 1 === preg_match('/[^\s\*].*\R.*[^\s\*]/s', $commentContent)) {
                 continue;
             }
+
             $nextTokenIndex = $index + 1;
             if (isset($tokens[$nextTokenIndex])) {
                 $nextToken = $tokens[$nextTokenIndex];
-                if (!$nextToken->isWhitespace() || !preg_match('/\R/', $nextToken->getContent())) {
+                if (!$nextToken->isWhitespace() || 1 !== preg_match('/\R/', $nextToken->getContent())) {
                     continue;
                 }
 
                 $tokens[$nextTokenIndex] = new Token([$nextToken->getId(), ltrim($nextToken->getContent(), " \t")]);
             }
 
             $content = '//';
-            if (preg_match('/[^\s\*]/', $commentContent)) {
-                $content = '// '.preg_replace('/[\s\*]*([^\s\*](.+[^\s\*])?)[\s\*]*/', '\1', $commentContent);
+            if (1 === preg_match('/[^\s\*]/', $commentContent)) {
+                $content = '// '.preg_replace('/[\s\*]*([^\s\*](?:.+[^\s\*])?)[\s\*]*/', '\1', $commentContent);
```

Comment:
I always thought raising the complexity of a regex would also slow it down, in this case is the opposite, thank you for the TIL
```php
$patterns = [
    'without-?:'    => '/[\s\*]*([^\s\*](.+[^\s\*])?)[\s\*]*/',
    'with-?:'       => '/[\s\*]*([^\s\*](?:.+[^\s\*])?)[\s\*]*/',
];

foreach ($patterns as $type => $pattern) {
    $start = microtime(true);
    for ($i = 0; $i < 10000000; ++$i) {
        preg_replace($pattern, '\1', ' * * a b c * * ');
    }
    $elapsed = microtime(true) - $start;

    var_dump(sprintf('%10s => %s', $type, $elapsed));
}

// string(29) "without-?: => 2.6824641227722"
// string(29) "   with-?: => 2.6746520996094"
```